ui/admin/app/routes/scopes/scope/session-recordings.js

```diff
@@ -68,4 +70,28 @@ export default class ScopesScopeSessionRecordingsRoute extends Route {
       };
     }
   }
+
+  /**
+   * Reapplies storage policy dates to session recording
+   * @param {SessionRecordingModel}
+   */
+  @action
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess(function () {
+    const { end_time } = this.modelFor(
+      'scopes.scope.session-recordings.session-recording',
+    );
+    return !end_time
+      ? 'resources.policy.messages.later'
+      : 'resources.policy.messages.reapply';
+  })
+  async reapplyStoragepolicy(sessionRecording) {
+    try {
+      await sessionRecording.reapplyStoragePolicy();
+    } catch (e) {
+      await this.router.replaceWith('scopes.scope.session-recordings');
+      this.refresh();
```

Comment:
Did we move this action just to call `this.refresh` on the session recordings list? Does `this.router.refresh()` or `this.router.refresh('scopes.scope.session-recordings')` not work?

Reply:
yeah, that's why it was moved, but `refresh` method works on the router, I didn't realize we could call `refresh` from the router like that, I had this on the model earlier. It has been moved back to the original spot now.